lib-injection/auto_inject.rb

```diff
@@ -2,22 +2,29 @@
 
 begin
   require 'open3'
-
-  failure_prefix = 'Datadog lib injection failed:'
   support_message = 'For help solving this issue, please contact Datadog support at https://docs.datadoghq.com/help/.'
 
-  _, status = Open3.capture2e({'DD_TRACE_SKIP_LIB_INJECTION' => 'true'}, 'bundle show ddtrace')
+  def debug_log(msg)
+    STDOUT.puts msg if ENV["DD_TRACE_DEBUG"] == "true"
+  end
 
+  _, status = Open3.capture2e({'DD_TRACE_SKIP_LIB_INJECTION' => 'true'}, 'bundle show ddtrace')
   if status.success?
-    STDOUT.puts '[ddtrace] ddtrace already installed... skipping auto-injection' if ENV['DD_TRACE_DEBUG'] == 'true'
+    debug_log '[ddtrace] ddtrace already installed... skipping injection'
     return
   end
 
   require 'bundler'
+  require "bundler/cli"
   require 'shellwords'
 
   if Bundler.frozen_bundle?
-    STDERR.puts "[ddtrace] #{failure_prefix} Cannot inject with frozen Gemfile, run `bundle config unset deployment` to allow lib injection. To learn more about bundler deployment, check https://bundler.io/guides/deploying.html#deploying-your-application. #{support_message}"
+    STDERR.puts "[ddtrace] Injection failed: Frozen `Gemfile` because Bundler is configured with `deployment`."
```

Comment:
What about the message to be clear that it's unable to inject?
[ddtrace] Injection failed: Unable to inject into a Frozen `Gemfile` (Bundler is configured with `deployment`) 

?

Reply:
Also will there be a guide explaining what to do for frozen Gemfiles?